tools/pydfu.py

```diff
@@ -573,6 +587,13 @@ def main():
     parser.add_argument(
         "-u", "--upload", help="read file from DFU device", dest="path", default=False
     )
+    parser.add_argument(
+        "-a",
+        "--address",
+        help="specify target memory address(hex or dec) when uploading .bin files",
+        dest="address",
+        default=False,
```

Comment:
You should be able to follow how the vid/pid arguments are done:
```py
parser.add_argument("-a", "--address", help="...", type=lambda x: int(x, 0), default=None)
```
and then use it like:
```py
if args.address is None:
    # not specified
else:
    args.address  # already an int type
```

Reply:
Solved by ab0266f1df631264d8edb4f243df18d21035b3f9